### PR TITLE
doil_into_container dic_335 add php container to handle doil commands

### PR DESCRIPTION
**Pull request by daniwe4**

### CODE REVIEW

File: app/src/Lib/FileSystem/Filesystem.php [hunk 1]
```diff
@@ -8,8 +8,10 @@ interface Filesystem
 {
     public const DOIL_PATH_MAIN_CONFIG = "/etc/doil";
     public const DOIL_PATH_LIB = "/usr/local/lib/doil";
-    public const DOIL_PATH_SHARE = "/usr/local/share/doil";
     public const DOIL_PATH_BIN = "/usr/local/bin/doil";
+    public const DOIL_PATH_SHARE = "/usr/local/share/doil";
+    public const DOIL_PATH_SHARE_STACK = "/usr/local/share/doil/stack";
```
Comment (tbongers-cat) [inline app/src/Lib/FileSystem/Filesystem.php:13]: The two added constants are not used anywhere, right?

File: app/src/Commands/System/UninstallCommand.php
Comment (tbongers-cat) [inline app/src/Commands/System/UninstallCommand.php:1]: The command needs to be removed from cli.php and the checklists in TESTING.md need to be adapted to the new uninstall script (no more -prune option).

**Review by tbongers-cat — CHANGES_REQUESTED** [review]
Nicely done, looking forward to using it =)
I have some small changes/questions and some annotations on translations/expressions (as you probably expected).

File: README.md [hunk 2]
```diff
@@ -3,13 +3,17 @@
 **doil** provides you with a simple way to create and manage development and
 testing environments for ILIAS. It will create and provision a docker container
 according to your requirements, pull the ILIAS version you want to use and even
-install it if possible.
+install it if possible. Also, **doil** provides you a mail server wich collects all
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply README.md:6]: Done!

File: README.md [hunk 3]
```diff
@@ -3,13 +3,17 @@
...
+mails from your instances, so you can test the ILIAS emailing.
 
 ## Installation
 
 1. download and unpack the [latest release](https://github.com/conceptsandtraining/doil/releases)
 1. cd into the unpacked directory
 1. if you run doil on a remote host ensure to change the host name in `setup/conf/doil.conf` to your host name
+1. adjust your mail password in `setup/conf/doil.conf`
+1. if you run global instances ensure to adjust 'global_instances_path' in `setup/conf/doil.conf` to specify
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply README.md:15]: Done!

File: README.md [hunk 4]
```diff
@@ -3,13 +3,17 @@
...
+   where to place them, default is '/srv/instances'. Attention, pathes with 'home' are not allowed here.
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply README.md:16]: Done!

File: README.md [hunk 5]
```diff
@@ -35,8 +39,11 @@ Otherwise, use the update script.
 
 1. checkout the newest master branch or copy and extract the zip
 2. cd into the unpacked directory
-3. execute sudo ./setup/update.sh in order to update **doil**
-4. you can remove the downloaded folder afterwards
+3. if you run global instances ensure to adjust 'global_instances_path' in `setup/conf/doil.conf` to specify 
+where to place them, default is '/srv/instances'. Attention, pathes with 'home' are not allowed here. The update
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply README.md:43]: Done!

File: README.md [hunk 6]
```diff
@@ -258,11 +265,19 @@ via Dockers volumes and can be accessed from the host system.
 
 **doil** comes with some helpers which are useful if you want to hack on **doil**:
 
-* `doil system:uninstall` will remove doil from your system. users, instance and config remain 
-* `doil system:uninstall --prune` will remove doil completely from your system for all users
 * `doil -V|--version` displays the version
 * `doil` displays the main help page
 
+Also there is a script to uninstall **doil** in the doil repo. If you have already delete
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply README.md:271]: Done!

File: setup/uninstall.sh [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+          ;;
+  esac
+elif [[ $1 == "-a" || $1 == "--all" ]]
+then
+  delete all
+elif [[ $1 == "-n" || $1 == "--not-all" ]]
+then
+  delete
+else
+  echo "Usage: $0 [-a|--all] | [-n|--not-all]"
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply setup/uninstall.sh:63]: Yes, we want to bypass the security question in the case we call it from another script.

I add a warning to the help text.
`echo "Please use the parameters carefully. There is no further security query."`

File: setup/system.sh [hunk 8]
```diff
@@ -166,19 +223,24 @@ function doil_system_setup_ip() {
   IPEXIST=$(grep "172.24.0.254" /etc/hosts)
   if [[ -z ${IPEXIST} ]]
   then
-    printf "172.24.0.254 doil" >> "/etc/hosts"
+    HOST=$(doil_get_conf host)
+    printf "172.24.0.254 ${HOST}" >> "/etc/hosts"
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply setup/system.sh:227]: Feature Ticket:
https://github.com/conceptsandtraining/doil/issues/344

But we do check if the network already exists. But if it exists doil will break. So its a nice feature request.

File: setup/system.sh [hunk 9]
```diff
@@ -52,89 +143,55 @@ function doil_system_add_user_to_doil_group() {
...
+  if [ ! -d /usr/local/lib/doil/server ]
   then
-    mkdir /usr/local/share/doil/stack
+    mkdir /usr/local/lib/doil/server
   fi
 
   if [ ! -d /usr/local/share/doil/instances ]
   then
-    mkdir /usr/local/share/doil/instances
+    mkdir -p /usr/local/share/doil/instances
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply setup/system.sh:169]: Thats right, but i will refactor that in a future PR.

File: setup/doil.sh [hunk 10]
```diff
@@ -5,14 +5,21 @@
...
-# Copyright (C) 2020 - 2021 Laura Herzog (laura.herzog@concepts-and-training.de)
+# Copyright (C) 2020 - 2023 Daniel Weise (daniel.weise@concepts-and-training.de)
 # Permission to copy and modify is granted under the AGPL license
 #
 # Contribute: https://github.com/conceptsandtraining/doil
 #
 # /ᐠ｡‸｡ᐟ\
 # Thanks to Concepts and Training for supporting doil
 
+doil_get_conf() {
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply setup/doil.sh:16]: The version from helper.sh will not be copied to the doil directory.

File: app/src/Lib/FileSystem/Filesystem.php (hunk 1 above)
Comment (daniwe4) [reply app/src/Lib/FileSystem/Filesystem.php:13]: Done!

File: app/src/Commands/Instances/DeleteCommand.php [hunk 11]
```diff
@@ -144,12 +144,6 @@ protected function deleteInstance(
         $this->docker->removeContainer($instance . "_" . $suffix);
 
         $this->docker->executeCommand(self::SALT_MAIN, "doil_saltmain", "salt-key", "-d", "$instance.$suffix", "-y", "-q");
-
-        $proxy_path = self::PROXY_SITES . $instance . ".conf";
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply app/src/Commands/Instances/DeleteCommand.php:148]: Done!

File: app/src/Commands/System/UninstallCommand.php
Comment (daniwe4) [reply app/src/Commands/System/UninstallCommand.php:1]: Done!

**Review by tbongers-cat — APPROVED** [review]
I approve!

**Pull request merged**